generator/graphql-kotlin-schema-generator/src/main/kotlin/com/expediagroup/graphql/generator/internal/types/generateDirective.kt

```diff
@@ -70,38 +80,40 @@ private fun getDirective(generator: SchemaGenerator, directiveInfo: DirectiveInf
         val directiveArguments: List<KProperty<*>> = directiveClass.getValidProperties(generator.config.hooks)
 
         directiveArguments.forEach { prop ->
-            val argument = generateDirectiveArgument(prop, directiveInfo, generator)
+            val argument = generateDirectiveArgument(prop, generator)
             builder.argument(argument)
         }
 
         builder.build()
     }
 
+    val appliedDirective = directive.toAppliedDirective()
     return if (directive.arguments.isNotEmpty()) {
-        // update args for this instance
-        val builder = GraphQLDirective.newDirective(directive)
-        directiveInfo.directive.annotationClass.getValidProperties(generator.config.hooks).forEach { prop ->
-            val defaultArgument = directive.getArgument(prop.name)
-            val value = prop.call(directiveInfo.directive)
-            val argument = GraphQLArgument.newArgument(defaultArgument)
-                .value(value)
-                .build()
-            builder.argument(argument)
+        appliedDirective.transform { builder ->
+            directiveInfo.directive.annotationClass.getValidProperties(generator.config.hooks).forEach { prop ->
+                val value = prop.call(directiveInfo.directive)
+                val directiveArgument = directive.getArgument(prop.name)
+                    .toAppliedArgument()
+                    .transform { argumentBuilder ->
+                        argumentBuilder.valueProgrammatic(value)
+                    }
+
+                builder.argument(directiveArgument)
+            }
         }
-        builder.build()
     } else {
-        directive
+        appliedDirective
     }
 }
 
-private fun generateDirectiveArgument(prop: KProperty<*>, directiveInfo: DirectiveInfo, generator: SchemaGenerator): GraphQLArgument {
+private fun generateDirectiveArgument(prop: KProperty<*>, generator: SchemaGenerator): GraphQLArgument {
     val propertyName = prop.name
-    val value = prop.call(directiveInfo.directive)
```

Comment:
This never worked for default values as it is called on an instance of the annotation. Unsure how it worked for federation before (guessing values were not printed in the sdl).